### PR TITLE
Feature/asset management

### PR DESCRIPTION
**Pull request by TornikeDz**
image
image

Well some parts of design may not be spot on.
Added data to Asset management Page:
- headings
- category switch
- add button(non-functional)
- assets table(reusable) with pagination

### CODE REVIEW

File: src/api/asset-management.api.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { getHttpRequest } from '@/services/httpService'
+
+export const fetchItemsByCategory = async asset => {
+  return await getHttpRequest(`../../public/api/asset-management/${asset}.json`)
+}
```
Comment (iv-thomson) [inline src/api/asset-management.api.js]: I like that you extracted it to a separate file. The path is a bit excessive though. See how others did it: https://github.com/iv-thomson/andersen-vue-study/blob/82f2c1f681a6c42c5c4b6fa91f57f0a5c008bb78/src/views/ServiceTicketReportingView.vue#L80

File: src/components/AssetsTable/AssetsTable.vue [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        <th
+          v-for="(column, index) of columns"
+          :key="index"
+          class="assets-table__header-item"
+          @click="sortTable(column.label)"
+          :class="{
+            sortable: column.sortable !== false,
+            asc: this.setSortType('asc', column),
+            desc: this.setSortType('desc', column),
+            'sort-active': sortConfig.column === column.label,
```
Comment (iv-thomson) [inline src/components/AssetsTable/AssetsTable.vue]: this is ok, but just FYI: https://vuejs.org/style-guide/rules-recommended.html#element-attribute-order
These recommendations will make your components a bit more readable

File: src/components/AssetsTable/AssetsTable.vue [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      const startIndex = (this.currentPage - 1) * this.rowsPerPage
+      const endIndex = startIndex + this.rowsPerPage
+      return sortedRows.slice(startIndex, endIndex)
+    },
+
+    sortRows(rows) {
+      if (!this.sortConfig.column) return rows
+
+      const { column, direction } = this.sortConfig
+      const sortedList = [...rows].sort((a, b) => {
```
Comment (iv-thomson) [inline src/components/AssetsTable/AssetsTable.vue]: you can use toSorted: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted

Comment (iv-thomson) [reply src/components/AssetsTable/AssetsTable.vue]: looks like you have sortRows method and `sortedRows` computed property that are seemingly doing the same thing. Could you please explain this a bit?

File: src/views/AssetManagementView.vue [hunk 4]
```diff
@@ -1,5 +1,172 @@
...
+            alt="right arrow"
+          />
+        </li>
+      </ul>
+    </div>
+    <div class="asset-management-container">
+      <CategorySwitch
+        :categories="categoryOptions"
+        :defaultCategory="selectedCategory"
+        @selectCategory="handleSelectedCategory"
```
Comment (iv-thomson) [inline src/views/AssetManagementView.vue]: you could've used v-model for handling selected category

**Review by iv-thomson — APPROVED** [review]

File: src/components/AssetsTable/AssetsTable.vue (hunk 3 above)
Comment (TornikeDz) [reply src/components/AssetsTable/AssetsTable.vue]: > looks like you have sortRows method and `sortedRows` computed property that are seemingly doing the same thing. Could you please explain this a bit?

I modified it and removed the second one. Previously first one was used to initialization mostly and second one was used for updating the vue. Originally I added sortedRows cuz some bug appeared after adding pagination. Changing pages caused table to lose sort ability after click.

**Pull request merged**